motoman_driver/MotoPlus/IoServer.c

```diff
@@ -427,19 +453,184 @@ int Ros_IoServer_WriteIOGroup(SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
 	replyMsg->header.msgType = ROS_MSG_MOTO_WRITE_IO_GROUP_REPLY;
 	replyMsg->header.commType = ROS_COMM_SERVICE_REPLY;
 
-	for (i = 0; i < 8; i += 1)
+	bAddressOk = Ros_IoServer_IsValidWriteAddress(receiveMsg->body.writeIOGroup.ioAddress, IO_ACCESS_GROUP);
+	bValueOk = Ros_IoServer_IsValidWriteValue(receiveMsg->body.writeIOGroup.ioValue, IO_ACCESS_GROUP);
+
+	if (bAddressOk && bValueOk)
+	{
+		for (i = 0; i < 8; i += 1)
+		{
+			ioWriteData[i].ulAddr = (receiveMsg->body.writeIOGroup.ioAddress * 10) + i;
+			ioWriteData[i].ulValue = (receiveMsg->body.writeIOGroup.ioValue & (1 << i)) >> i;
+		}
+		apiRet = mpWriteIO(ioWriteData, 8);
```

Comment:
Should this be updated as well to use the constant(s)?

Reply:
You're right. I just missed that.